src/main/java/net/sf/jabref/logic/formatter/casechanger/Title.java

```diff
@@ -16,6 +17,11 @@ public Title(String title) {
         this.words.addAll(new TitleParser().parse(title));
     }
 
+    public List<Word> getWordsReadOnly() {
```

Comment:
Why not immediately replace `getWords` with `getWordsReadOnly`, i.e. just have `getWords` return `Collections.unmodifiableList(words);` and skip the introduction of `getWordsReadOnly`? From what I see this should not break the existing implementations.